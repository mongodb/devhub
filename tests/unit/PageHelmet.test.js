import React from 'react';
import { shallow } from 'enzyme';
import PageHelmet from '~components/dev-hub/page-helmet';

const siteUrl = 'https://developer.mongodb.com';
const title = 'Standard title';

jest.mock('../../src/hooks/use-site-metadata', () => ({
    useSiteMetadata: () => ({ siteUrl, title }),
}));

describe('PageHelmet', () => {
    it('renders standard correctly', () => {
        const tree = shallow(<PageHelmet />);
        expect(tree).toMatchSnapshot();
    });

    it('renders with custom props correctly', () => {
        const tree = shallow(
            <PageHelmet
                title="Test Title"
                pagePath="/test-url.com"
                description="test description"
                type="test prop"
            />
        );
        expect(tree).toMatchSnapshot();
    });
});
