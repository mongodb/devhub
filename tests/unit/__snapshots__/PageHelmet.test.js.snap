// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHelmet renders standard correctly 1`] = `
<SEO
  articleTitle="Standard title"
  canonicalUrl="https://developer.mongodb.com"
  image="https://developer.mongodb.com/public/images/social_share_generic.png"
  metaDescription="Code, content, tutorials, programs and community to enable developers of all skill levels on the MongoDB Data Platform which includes Atlas, Realm, Compass, Data Lake and more. Whether you're coding in Java, JavaScript, C#, Python, Node, Go or looking for how this fits with IOT, AI, ML - join or follow us here."
  ogDescription="Code, content, tutorials, programs and community to enable developers of all skill levels on the MongoDB Data Platform which includes Atlas, Realm, Compass, Data Lake and more. Whether you're coding in Java, JavaScript, C#, Python, Node, Go or looking for how this fits with IOT, AI, ML - join or follow us here."
  ogTitle="Standard title"
  ogUrl="https://developer.mongodb.com"
  title="Standard title"
  twitter={
    Object {
      "description": "Code, content, tutorials, programs and community to enable developers of all skill levels on the MongoDB Data Platform which includes Atlas, Realm, Compass, Data Lake and more. Whether you're coding in Java, JavaScript, C#, Python, Node, Go or looking for how this fits with IOT, AI, ML - join or follow us here.",
      "image": "https://developer.mongodb.com/public/images/social_share_generic.png",
      "title": "Standard title",
    }
  }
  type="website"
/>
`;

exports[`PageHelmet renders with custom props correctly 1`] = `
<SEO
  articleTitle="Test Title"
  canonicalUrl="https://developer.mongodb.com/test-url.com/"
  image="https://developer.mongodb.com/public/images/social_share_generic.png"
  metaDescription="test description"
  ogDescription="test description"
  ogTitle="Test Title"
  ogUrl="https://developer.mongodb.com/test-url.com/"
  title="Test Title"
  twitter={
    Object {
      "description": "test description",
      "image": "https://developer.mongodb.com/public/images/social_share_generic.png",
      "title": "Test Title",
    }
  }
  type="test prop"
/>
`;
